[OSDOCS-5660] Added additional context to ROSA HCP documentation (#58031)
openshift/openshift-docs: rosa_hcp/rosa-hcp-sts-creating-a-cluster-quickly.adoc

If you are looking for a quickstart guide for ROSA, see xref:../rosa_getting_started/rosa-quickstart-guide-ui.adoc#rosa-quickstart-guide-ui[{product-title} quickstart guide].
====

> {hcp-first} for {product-title} (ROSA) offers another method for creating ROSA clusters. With {hcp} for ROSA, each cluster has a dedicated control plane that is isolated in Red Hat’s accounts. 

Reviewer: ```suggestion
{hcp-first} for {product-title} (ROSA) offers a more efficient and reliable architecture for creating ROSA clusters. With {hcp} for ROSA, each cluster has a dedicated control plane that is isolated in a Red Hat account. 
```

[IMPORTANT]
====

> You must create a new cluster to use {hcp} for {product-title} functionality. Each instance of {hcp} for ROSA is limited to a single cluster.

Reviewer: ```suggestion
It is not possible to "upgrade" to a {hcp-first} architecture, you must create a new cluster to benefit from the {hcp} for {product-title} functionality.
```

* link:https://docs.aws.amazon.com/vpc/latest/userguide/vpc-getting-started.html[Get Started with Amazon VPC]
* link:https://developer.hashicorp.com/terraform[HashiCorp Terraform documentation]

[id="rosa-hcp-byo-odic-overview_{context}"]

> == Bring-Your-Own OpenID Connect Configuration

Reviewer 2: I am not sure if the "bring your own" is precise
@gdbranco Would you help double confirm if we can use "Bring-Your-Own" or byo in doc?

Reviewer 3: I'd say we should probably move external references to this as Unmanaged OIDC Configuration (when it is hosted in customers aws account) and Managed OIDC Configuration (when hosted by Red Hat)

@wgordon17 Do you have any insights on correct terms?

Author: @yingzhanredhat, good point. I revised this to say `Creating a managed OpenID Connect Configuration`. Then, the first level module is `Generating your own OpenID Connect configuration`. Does this work better?

CC: @gdbranco @wgordon17

> When using a Red-Hat hosted cluster, you can create an OpenID Connect (OIDC) provider that is generated by the CLI and stored within Red Hat's AWS account. This process does not provide a private key for users to access.

Reviewer 2: I think it should be OpenID Connect (OIDC) **config**. And another issue is the un-managed OIDC config is not stored in the Red Hat account. @gdbranco Would you help double confirm it too?

> you can create an OpenID Connect (OIDC) provider that is generated by the CLI and stored within Red Hat's AWS account.

Reviewer 3: I think we need some session to properly address both types of Managed vs Unmanaged

> When using a Red-Hat hosted cluster, you can create a managed or unmanaged OpenID Connect (OIDC) configuration that is generated by the CLI. With a managed OIDC configuration, it is stored within Red Hat's AWS account. A generated unmanaged OIDC configuration is stored in your AWS account and registers the intention to use that configuration with OCM. This generated OIDC configuration provides the private key for you to access. This process does not provide a private key for users to access.

Reviewer 4: Should this be "Red Hat"? Or maybe you meant "Red Hat-hosted"? Also suggestion to include "unmanaged" here because its kind of confusing to parse which one is being referred to:
```suggestion
When using a Red Hat hosted cluster, you can create a managed or unmanaged OpenID Connect (OIDC) configuration that is generated by the CLI. With a managed OIDC configuration, it is stored within Red Hat's AWS account. A generated unmanaged OIDC configuration is stored in your AWS account and registers the intention to use that configuration with OCM. This generated unmanaged OIDC configuration provides the private key for you to access. This process does not provide a private key for users to access.
```
also, does the phrase "registers the intention to use" mean something specific? It is also a little confusing to understand what is meant here
